segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -333,14 +336,8 @@ protected int getMaxCompactionSize() {
 
     private <T> CompletableFuture<Long> commit(Collection<T> toCommit, Function<Collection<T>, BufferView> serializer,
                                                DirectSegmentAccess segment, long tableSegmentOffset, Duration timeout) {
-        if (tableSegmentOffset == NULL_TABLE_SEGMENT_OFFSET || tableSegmentOffset == segment.getInfo().getLength()) {
             BufferView s = serializer.apply(toCommit);
-            return segment.append(s, null, timeout);
-        } else {
-            CompletableFuture<Long> future = new CompletableFuture<>();
-            future.completeExceptionally(new BadOffsetException(segment.getInfo().getName(), segment.getInfo().getLength(), tableSegmentOffset));
-            return future;
-        }
+            return segment.append(s, null, tableSegmentOffset, timeout);
```

Comment:
I was only asking you to do the conditional commit for the second case. Please bring back the `if-else` block and only pass the explicit offset in the `else` case. Do not pass `tableSegmentOffset` in the `if` case as it's just a coincidence that `-1` is the "no-offset" append that is now in the Segment Store. By using the other `append` (non-conditional), you shield yourself from any future change in the segment store that may affect how that works.